### PR TITLE
Handle stream error in StandaloneConnector.js

### PR DESCRIPTION
**Pull request by peteraw77**
Solution proposed by @OPScorp in #1209 . Solves #1235 as well. I'm not sure why he didn't PR this himself but it was helpful for myself and others so I would like to get it merged.

### CODE REVIEW

**Comment by AVVS** [comment]
@peteraw77 thanks for the PR, but it does definitely need a test case. Error handlers are typically assigned later on in the chain once the connection was created. I dont remember the reasoning for not having an error handler in the create helper, either way a test covering this would be required to get this merged

**Comment by peteraw77** [comment]
@AVVS where would be the right place to put the test case?

**Comment by peteraw77** [comment]
Added a test in test/functional/connection.ts but I'm having a hard time getting the tests to reproduce the error. This was my best attempt but it's definitely not ideal.

File: test/functional/connection.ts [hunk 1]
```diff
@@ -240,6 +240,21 @@ describe("connection", function () {
         done();
       });
     });
+
+    it("should reject on initial connection error", function (done) {
+      const stub = sinon
+        .stub(net.Socket.prototype, "connect")
+        .callsFake((options) => {
+          net.Socket.prototype.emit("error");
```
Comment (AVVS) [inline test/functional/connection.ts:248]: this need to be async, .connect() doesnt throw in the same tick - wrap this in process.nextTick

https://nodejs.org/api/net.html#net_socket_connect

Comment (peteraw77) [reply test/functional/connection.ts]: Thanks, it seems to be capturing the behaviour I intended it to now

**Comment by francescov1** [comment]
Is there an expected timeline on when this will be merged? This issue has been causing endless headaches for our team

**Comment by drobnikj** [comment]
Ping here. I would be great to merge this fix soon. It generates Unhandled errors, which are hard to debug. In our case it was an error: 
```
[ioredis] Unhandled error event: Error: read EFAULT
    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)
    at TCP.callbackTrampoline (internal/async_hooks.js:126:14)
```
It takes us time to find that bug comes from this issue.
Thanks!

**Comment by francescov1** [comment]
> Ping here. I would be great to merge this fix soon. It generates Unhandled errors, which are hard to debug. In our case it was an error:
> 
> ```
> [ioredis] Unhandled error event: Error: read EFAULT
>     at TCP.onStreamRead (internal/stream_base_commons.js:209:20)
>     at TCP.callbackTrampoline (internal/async_hooks.js:126:14)
> ```
> 
> It takes us time to find that bug comes from this issue.
> Thanks!

Same here, it's such a simple, obvious but crucial fix...

**Comment by kapalkat** [comment]
Please merge that change as we had to write dedicated service to catch those unhandled errors...

**Comment by ctrlaltdylan** [comment]
Also pinging! If I was a maintainer I would merge. But alas I am a simple consumer of this great package.

**Comment by luin** [comment]
Sorry for the late response. I didn't notice this PR... Will take a look soon!

**Comment by luin** [comment]
`reject(err)` here actually always does nothing as the Promise has already resolved at line 83. This is equivalent to `this.stream.once('error', () => {})`, all it does is ignoring the stream error instead of handling them properly. So this PR should be considered as a workaround.

I created a PR to address the root issue: #1296. I'll leave this open and may merge this if I can't get #1296 done very soon. Thank you for the contributions!

**Comment by ctrlaltdylan** [comment]
You the man @luin !

**Pull request closed without merging**